### ISSUE TITLE
Unable to build in release

### ISSUE THREAD

**Issue by ChobitsTheZero** [TorchCraft/TorchCraft#301]
Have some issues when trying to build the BMEnv Solution. Get a ton of different errors that I am unsure how to solve. I followed the steps for developing the TorchCraft Server, although I may have misinterpreted the step "Clone TorchCraft into $STARCRAFT/TorchCraft.".

I simply copied the TorchCraft folder into the Starcraft folder where I have bwapi located.

Here is a screenshot of the multiple errors that I recieved.

![image](https://user-images.githubusercontent.com/25343961/53648130-7086cf00-3c3f-11e9-9e25-d45a2477da24.png)

**Comment by dgant** [comment]
It looks like there's an issue importing the types specified in https://github.com/TorchCraft/TorchCraft/blob/master/BWEnv/fbs/messages_generated.h -- is that file present (in the correct location), or there an error about the import failing?

**Comment by ChobitsTheZero** [comment]
> It looks like there's an issue importing the types specified in https://github.com/TorchCraft/TorchCraft/blob/master/BWEnv/fbs/messages_generated.h -- is that file present (in the correct location), or there an error about the import failing?

Yes, the file "messages_generated.h" is present in the folder fbs.

Dosn't look like there are any errors about import failiure

![image](https://user-images.githubusercontent.com/25343961/53653768-599aa980-3c4c-11e9-8a74-ca54c7320f64.png)

![image](https://user-images.githubusercontent.com/25343961/53653778-615a4e00-3c4c-11e9-9b41-173975ae62b8.png)

**Comment by ChobitsTheZero** [comment]
Not sure if "cloning" is the same as "copying" since I just copied the entire TorchCraft folder into the Starcraft folder with bwapi inside it.

**Comment by dgant** [comment]
Cloning refers to using `git clone` to copy the repository plus its history, allowing you to perform Git operations on it. I'd recommend using it but it should be unrelated to your issue.

With some further research, it looks like the issue is the compiler not recognizing the `noexcept` keyword: https://en.cppreference.com/w/cpp/language/noexcept_spec . The Flatbuffers library uses `noexcept` when it believes the C++ compiler is of a version which understands it: https://github.com/TorchCraft/TorchCraft/blob/master/BWEnv/fbs/base.h#L123

It's possible that the Flatbuffers macro is incorrectly guessing that you're using a MSVC version that supports `noexcept`. One way you could test this is to replace https://github.com/TorchCraft/TorchCraft/blob/master/BWEnv/fbs/base.h#L125 with `#define FLATBUFFERS_NOEXCEPT` and see if that changes the errors you see.

**Comment by ChobitsTheZero** [comment]
> Cloning refers to using `git clone` to copy the repository plus its history, allowing you to perform Git operations on it. I'd recommend using it but it should be unrelated to your issue.
> 
> With some further research, it looks like the issue is the compiler not recognizing the `noexcept` keyword: https://en.cppreference.com/w/cpp/language/noexcept_spec . The Flatbuffers library uses `noexcept` when it believes the C++ compiler is of a version which understands it: https://github.com/TorchCraft/TorchCraft/blob/master/BWEnv/fbs/base.h#L123
> 
> It's possible that the Flatbuffers macro is incorrectly guessing that you're using a MSVC version that supports `noexcept`. One way you could test this is to replace https://github.com/TorchCraft/TorchCraft/blob/master/BWEnv/fbs/base.h#L125 with `#define FLATBUFFERS_NOEXCEPT` and see if that changes the errors you see.

Dosn't seem like it changed anything, still the same errors regardless of the change to 
#define FLATBUFFERS_NOEXCEPT

**Comment by dgant** [comment]
Even the "'noexcept': unknown override specifier" errors?

**Comment by ChobitsTheZero** [comment]
> Even the `"'noexcept': unknown override specifier"` errors?

Yes, as shown in the picture below, this is with the change in action.

![image](https://user-images.githubusercontent.com/25343961/53655370-4b4e8c80-3c50-11e9-884e-1965a60d56ef.png)

![image](https://user-images.githubusercontent.com/25343961/53655384-530e3100-3c50-11e9-94de-1fa674c28c05.png)

**Comment by ChobitsTheZero** [comment]
Clicking on the errors leads to this line "Frame& operator=(Frame other) noexcept;" in frame.h

**Comment by ChobitsTheZero** [comment]
> Clicking on the errors leads to this line `Frame& operator=(Frame other) noexcept;` in frame.h

**Comment by ChobitsTheZero** [comment]
Clicking on the errors leads to this line `Frame& operator=(Frame other) noexcept;` in frame.h

**Comment by dgant** [comment]
That could indicate an older compiler version. What version of Visual Studio are you using? What version of MSVC++?

**Comment by ChobitsTheZero** [comment]
> That could indicate an older compiler version. What version of Visual Studio are you using? What version of MSVC++?

Ah right, that's correct. It was set at 2013 version for some reason. Now all I need to do is locate BWAPIClient.lib, not sure if this is required or if I can just scrap this lib all together. Since I am unable to locate it in the BWAPI folder.

**Comment by dgant** [comment]
I'm not familiar with BWAPIClient.lib, but if it exists I'd expect it to come with your distribution of BWAPI.

**Comment by ChobitsTheZero** [comment]
Ah, I believe the BWAPIClient.lib is for when using AIClient which I dont think i'll be doing.
